packages/infra/lib/api-stack/cq-directory-rebuilder.ts

```diff
@@ -0,0 +1,72 @@
+import { Duration } from "aws-cdk-lib";
+import { SnsAction } from "aws-cdk-lib/aws-cloudwatch-actions";
+import { IVpc } from "aws-cdk-lib/aws-ec2";
+import { Function as Lambda, Runtime } from "aws-cdk-lib/aws-lambda";
+import { Construct } from "constructs";
+import { EnvConfig } from "../../config/env-config";
+import { getConfig } from "../shared/config";
+import { LambdaLayers } from "../shared/lambda-layers";
+import { createScheduledLambda } from "../shared/lambda-scheduled";
+
+export type CqDirectoryRebuilderProps = {
+  stack: Construct;
+  lambdaLayers: LambdaLayers;
+  vpc: IVpc;
+  apiAddress: string;
+  alarmSnsAction?: SnsAction;
+};
+
+function getSettings(
+  props: CqDirectoryRebuilderProps,
+  config: NonNullable<EnvConfig["cqDirectoryRebuilder"]>
+) {
+  return {
+    ...props,
+    name: "ScheduledCqDirectoryRebuilder",
+    lambdaMemory: 256,
+    lambdaTimeout: Duration.seconds(900), // How long can the lambda run for, max is 900 seconds (15 minutes)
+    runtime: Runtime.NODEJS_18_X,
+    scheduleExpression: config.scheduleExpressions, // See: https://docs.aws.amazon.com/lambda/latest/dg/services-cloudwatchevents-expressions.html
+    url: `http://${props.apiAddress}/internal/carequality/directory/rebuild`,
+    httpTimeout: Duration.seconds(3),
```

Comment:
Maybe 5s to be safer? Sometimes the API runs a bit hot and it we won't lose anything by having a longer timeout, right?